base/iddict.jl

```diff
@@ -176,16 +193,17 @@ function get(default::Callable, d::IdDict{K,V}, @nospecialize(key)) where {K, V}
 end
 
 function get!(default::Callable, d::IdDict{K,V}, @nospecialize(key)) where {K, V}
-    val = ccall(:jl_eqtable_get, Any, (Any, Any, Any), d.ht, key, secret_table_token)
-    if val === secret_table_token
+    keyindex, inserted = ht_keyindex!(d, key)
+
+    if inserted
         val = default()
```

Comment:
It is not legal to use `_setindex` after this call

Reply:
You mean the call to `default` right? Presumably it's not legal because the dictionary might be mutated within the call?
I'll add a test for that then and revert the changes to this function.

Reply:
Yes, thanks